src/en/asurascans/src/eu/kanade/tachiyomi/extension/en/asurascans/AsuraScans.kt

```diff
@@ -297,16 +297,18 @@ class AsuraScans : ParsedHttpSource(), ConfigurableSource {
             .joinToString("") { it.data().substringAfter("\"").substringBeforeLast("\"") }
         val pagesData = PAGES_REGEX.find(scriptData)?.groupValues?.get(1) ?: throw Exception("Failed to find chapter pages")
         val pageList = json.decodeFromString<List<PageDto>>(pagesData.unescape()).sortedBy { it.order }
-        return pageList.mapIndexed { i, page ->
-            val newUrl = page.url.toHttpUrlOrNull()?.run {
-                newBuilder()
-                    .fragment("pageListParse")
-                    .build()
-                    .toString()
-            }
+        return pageList
+            .filterIndexed { i, _ -> !(preferences.hideCoverPages() && i == 0) }
+            .mapIndexed { i, page ->
+                val newUrl = page.url.toHttpUrlOrNull()?.run {
+                    newBuilder()
+                        .fragment("pageListParse")
+                        .build()
+                        .toString()
+                }
 
-            Page(i, imageUrl = newUrl ?: page.url)
-        }
+                Page(i, imageUrl = newUrl ?: page.url)
+            }
     }
```

Comment:
```suggestion
        val pageListFiltered = if (preferences.hideCoverPages()) { pageList.drop(0) } else { pageList }
        return pageListFiltered.mapIndexed { i, page ->
                val newUrl = page.url.toHttpUrlOrNull()?.run {
                    newBuilder()
                        .fragment("pageListParse")
                        .build()
                        .toString()
                }

                Page(i, imageUrl = newUrl ?: page.url)
            }
}
```
avoid double iteration

Reply:
What if pageList has 0 size?